remotespark/livyclientlib/sparkcontroller.py

```diff
@@ -72,6 +72,7 @@ def add_session(self, name, connection_string, skip_if_exists, properties):
         session.start()
         livy_client = self.client_factory.build_client(session)
         self.client_manager.add_client(name, livy_client)
+        livy_client.start()
```

Comment:
I don't know what the interface would look like.  We would need a new method for factories, `start_client(client)`, since the factory can't start the client for us in the `build_client` method.  Are you talking about some other design?